### PR TITLE
added example, additional info on configs and info on log rotation

### PR DESCRIPTION
**Pull request by ZahidMirza95**
**Description:** <Describe what has changed.>
Added example on how to use filelog receiver with log files. Also, added some details to the configuration descriptions. Additionally, added information on log rotation.

**Link to tracking Issue:** [23277](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23277)

**Testing:** N/A

**Documentation:** 

- Added example on how to use filelog receiver with log files
- Also, added some details to the configuration descriptions
- Additionally, added information on log rotation

### CODE REVIEW

File: receiver/filelogreceiver/README.md [hunk 1]
```diff
@@ -19,35 +19,35 @@ Tails and parses logs from files.
...
-| `encoding`                          | `utf-8`                              | The encoding of the file being read. See the list of supported encodings below for available options                                                                                                                                                            |
+| `include`                           | required                             | A list of file glob patterns that match the file paths to be read. The file path refers to the file path in the docker container (if you want to use a file from your local computer, you need to use a [docker volume](https://docs.docker.com/storage/volumes/)).                                                                                                                                                                                             |
```
Comment (rapphil) [inline receiver/filelogreceiver/README.md]: The collect will will not always run inside a container. I think we can omit this comment as about the file path inside the container as this has to do with the system and is general knowledge when using containers, you need to mount a volume with a path to the host if you need access to it.

File: receiver/filelogreceiver/README.md [hunk 2]
```diff
@@ -19,35 +19,35 @@ Tails and parses logs from files.
...
 | `include_file_path_resolved`        | `false`                              | Whether to add the file path after symlinks resolution as the attribute `log.file.path_resolved`.                                                                                                                                                               |
-| `poll_interval`                     | 200ms                                | The duration between filesystem polls                                                                                                                                                                                                                           |
+| `poll_interval`                     | 200ms                                | The duration between filesystem polls. Takes `time.Duration` as value.                                                                                                                                                                                          |
```
Comment (rapphil) [inline receiver/filelogreceiver/README.md]: What do you mean `time.Duration`?  Assume the person using this component does knowledge about go. 
Can we maybe replace this by 
> The duration between filesytem pools. The duration can include the unit. e.g.: 200ms, 1s, 1m.

What happens if the unit is not specified?

File: receiver/filelogreceiver/README.md [hunk 3]
```diff
@@ -19,35 +19,35 @@ Tails and parses logs from files.
...
-| `resource`                          | {}                                   | A map of `key: value` pairs to add to the entry's resource                                                                                                                                                                                                      |
-| `operators`                         | []                                   | An array of [operators](../../pkg/stanza/docs/operators/README.md#what-operators-are-available). See below for more details                                                                                                                                     |
+| `delete_after_read`                 | `false`                              | If `true`, each log file will be read and then immediately deleted. Requires that the `filelog.allowFileDeletion` feature gate is enabled. Cannot use if `start_at` is set to "end".                                                                            |
```
Comment (rapphil) [inline receiver/filelogreceiver/README.md]:
```suggestion
| `delete_after_read`                 | `false`                              | If `true`, each log file will be read and then immediately deleted. Requires that the `filelog.allowFileDeletion` feature gate is enabled. This option cannot be used if `start_at` is set to "end".                                                                            |
```

Comment (rapphil) [reply receiver/filelogreceiver/README.md]: Also, this option is ignored or will the code error if `start_at` is set to "end"?

File: receiver/filelogreceiver/README.md (hunk 2 above)
Comment (ZahidMirza95) [reply receiver/filelogreceiver/README.md]: Sure I can make those changes. As for when it's not specified, it'll default to nanoseconds. I'll add that info to the doc as well

File: receiver/filelogreceiver/README.md (hunk 3 above)
Comment (ZahidMirza95) [reply receiver/filelogreceiver/README.md]: An exception will be raised. Should this be documented as well?

File: receiver/filelogreceiver/README.md [hunk 4]
```diff
@@ -112,3 +121,26 @@ receivers:
...
+## Example - Reading from a log file
+
+Receiver Configuration
+```yaml
+receivers:
+  filelog:
+    include: [ /simple.log ]
+    operators:
+      - type: regex_parser
+        regex: '^(?P<time>\d{4}-\d{2}-\d{2}) (?P<sev>[A-Z]*) (?P<msg>.*)$'
```
Comment (rapphil) [inline receiver/filelogreceiver/README.md]: can we use timestamp in the format from the previous example?

File: receiver/filelogreceiver/README.md (hunk 3 above)
Comment (rapphil) [reply receiver/filelogreceiver/README.md]: yes, if that will crash teh collector, you can say something like: This option causes an error if it is used when `start_at` is set to "end".

File: receiver/filelogreceiver/README.md [hunk 5]
```diff
@@ -98,6 +98,15 @@ The header lines are not emitted by the receiver.
 
 Many parsers operators can be configured to embed certain followup operations such as timestamp and severity parsing. For more information, see [complex parsers](../../pkg/stanza/docs/types/parsers.md#complex-parsers).
 
+### Log Rotation
+
+Log Rotation is supported for File Log Receiver, but the following settings need to be configured:
```
Comment (rapphil) [inline receiver/filelogreceiver/README.md:103]: Nice work documenting this :)

**Review by rapphil — APPROVED** [review]

**Pull request merged**